src/drivers/amqp_interop/Queue.php

```diff
@@ -0,0 +1,383 @@
+
+        $producer = $this->context->createProducer();
+
+        if ($delay) {
+            $message->setProperty(self::H_DELAY, $delay);
+            $producer->setDeliveryDelay($delay * 1000);
+        }
+
+        if ($priority) {
+            $message->setProperty(self::H_PRIORITY, $priority);
+            $producer->setPriority($priority);
+        }
+
+        $producer->send($topic, $message);
+
+        return $message->getMessageId();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function status($id)
+    {
+        throw new NotSupportedException('Status is not supported in the driver.');
+    }
+
+    /**
+     * Opens connection and channel
+     */
+    protected function open()
+    {
+        if ($this->context) {
+            return;
+        }
+
+        switch ($this->driver) {
+            case 'lib':
+                $connectionClass = AmqpLibConnectionFactory::class;
+
+                break;
+            case 'ext':
+                $connectionClass = AmqpExtConnectionFactory::class;
+
+                break;
+            case 'bunny':
+                $connectionClass = AmqpBunnyConnectionFactory::class;
+
+                break;
+
+            default:
+                throw new \LogicException(sprintf('The given driver "%s" is not supported. Supported are "%s"', $this->driver, implode('", "', $this->supportedDrivers)));
+        }
+
+        $config = [
+            'dsn' => $this->dsn,
+            'host' => $this->host,
+            'port' => $this->port,
+            'user' => $this->user,
+            'pass' => $this->password,
+            'vhost' => $this->vhost,
+            'read_timeout' => $this->readTimeout,
+            'write_timeout' => $this->writeTimeout,
+            'connection_timeout' => $this->connectionTimeout,
+            'heartbeat' => $this->heartbeat,
+            'persisted' => $this->persisted,
+            'lazy' => $this->lazy,
+            'qos_global' => $this->qosGlobal,
+            'qos_prefetch_size' => $this->qosPrefetchSize,
+            'qos_prefetch_count' => $this->qosPrefetchCount,
+            'ssl_on' => $this->sslOn,
+            'ssl_verify' => $this->sslVerify,
+            'ssl_cacert' => $this->sslCacert,
+            'ssl_cert' => $this->sslCert,
+            'ssl_key' => $this->sslKey,
+        ];
+
+        $config = array_filter($config, function($value) {
+            return null !== $value;
+        });
+
+        /** @var AmqpConnectionFactory $factory */
+        $factory = new $connectionClass($config);
+
+        $this->context = $factory->createContext();
+
+        if ($this->context instanceof DelayStrategyAware) {
+            $this->context->setDelayStrategy(new RabbitMqDlxDelayStrategy());
+        }
+    }
+
+    public function setupBroker()
+    {
+        $queue = $this->context->createQueue($this->queueName);
+        $queue->addFlag(AmqpQueue::FLAG_DURABLE);
+        $queue->setArguments(['x-max-priority' => 4]);
```

Comment:
What is 4? Why 4?

Reply:
You can set a priority between 0 and 4. Here's the [priorities doc](https://www.rabbitmq.com/priority.html)

```
Resource Usage Considerations

If priority queues are desired, we recommend using between 1 and 10. Currently using more priorities will consume more resources (Erlang processes).
```

```
There is some in-memory and on-disk cost per priority level per queue. There is also an additional CPU cost, especially when consuming, so you may not wish to create huge numbers of levels.
The message priority field is defined as an unsigned byte, so in practice priorities should be between 0 and 255.
```

I add an ability to configure it and set the default value to 10 (as suggested by off doc). Added an exception for the case where a developer wants to use a higher value.